File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const [n, k] = input[0].split(" ").map(Number)
const bags = input.slice(1).map((row) => row.split(" ").map(Number))
const dp = Array.from({ length: k + 1 }, () => 0)

for (let [w, v] of bags) {
  for (let start = k; start >= w; start--) {
    if (dp[start] < dp[start - w] + v) {
      dp[start] = dp[start - w] + v
    }
  }
}
console.log(Math.max(...dp))

File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.py
import sys

input = sys.stdin.readline

n,k = map(int,input().split(' '))
dp = [0] * (k+1)

for i in range(n):
    w,v = map(int,input().split(' '))
    for j in range(k,w-1,-1):
        if(dp[j] < dp[j-w]+v):
            dp[j] = dp[j-w] +v
print(max(dp))